main.py

```diff
@@ -47,59 +47,62 @@ def run_one_generate(input_file_name : str, output_xml_file : str):
+    log_level = getattr(logging, "INFO", None)
+    logging.basicConfig(level=log_level)
+
+    # define the argument list of the main function
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("--input-config", default="./examples/2/topology.yaml", help="Name of the input YML file")
+    parser.add_argument("--output-xml", default="test.xml", help="Output precice XML config file")
+    parser.add_argument("--test-all", default="false", help="If this is true then we run in test mode and we generate all XMLs and we compare them")
+
+    # parsing the arguments into fields
+    args = parser.parse_args()
+    input_file_name_arg = args.input_config
+    output_xml_file_arg = args.output_xml
+    test_all = args.test_all
+
+    # test if we are in test mode
+    if test_all == "true":
+        # get all directories from the example directory and for each call the method
+        logging.info("Execute all test cases ... ")
+        # ./examples/*/topology.yaml
+        dirs = os.listdir("./examples/")
+        print( dirs )
+        for dir in dirs :
+            in_yaml_file_name = "./examples/" + dir + "/topology.yaml"
+            file_exists = os.path.isfile(in_yaml_file_name)
+            if file_exists:
+                # if this YAML file exists then call the function for it
+                ref_file_name = "./testing/output/" + dir + "_out_controller.xml"
+                out_file_name = "./testing/output/" + dir + "_out_controller_tmp.xml"
+                run_one_generate(in_yaml_file_name, out_file_name)
+                # once we generated the out file then we compare it
+                refernce = os.stat(ref_file_name)
+                test = os.stat(out_file_name)
+                #dom2 = lxml.etree.parse(out_file_name).getroot()
+                #comparison = compare(dom1, dom2)
+                if (refernce.st_size == test.st_size):
+                    # the two files are equal
+                    logging.info("Test " +  dir + " OK ")
+                    #overwrite the file
+                    shutil.copyfile(out_file_name, ref_file_name)
+                    os.unlink(out_file_name)
+                    pass
+                else:
+                    # size differe, jus issue an ERROR
+                    logging.info("Test " + dir + " FAILED ")
+                    pass
                 pass
-            else:
-                # size differe, jus issue an ERROR
-                logging.info("Test " + dir + " FAILED ")
-                pass
-            pass
-    logging.info("Finished all testing ... ")
-else:
-    # just call with the default arguments
-    run_one_generate(input_file_name_arg, output_xml_file_arg)
-    pass
-
+        logging.info("Finished all testing ... ")
+    else:
+        # just call with the default arguments
+        run_one_generate(input_file_name_arg, output_xml_file_arg)
+        pass
+
+if __name__ == "__main__":
+    main()
```

Comment:
I am not really sure how its going to look like in the future, ie if the name will have to stay main or if its going have to change but for time beeing we can do it like this.